### PR TITLE
show agent connection error when provisioning(NO MORE can not build dialer)

### PR DESCRIPTION
**Pull request by StrongMonkey**

### CODE REVIEW

**Comment by StrongMonkey** [comment]
need https://github.com/rancher/types/pull/416

File: pkg/controllers/management/clusterprovisioner/provisioner.go [hunk 1]
```diff
@@ -460,6 +460,10 @@ func (p *Provisioner) reconcileRKENodes(clusterName string) ([]v3.RKEConfigNode,
 			continue
 		}
 
+		if !v3.NodeConditionDeployed.IsTrue(machine) && !isCustom(machine) {
```
Comment (ibuildthecloud) [inline pkg/controllers/management/clusterprovisioner/provisioner.go]: I don't like this condition.  Let me think about it

Comment (StrongMonkey) [reply pkg/controllers/management/clusterprovisioner/provisioner.go]: agentDeployed? :)

**Comment by StrongMonkey** [comment]
https://github.com/rancher/rancher/issues/15035

File: pkg/controllers/management/clusterprovisioner/provisioner.go (hunk 1 above)
Comment (ibuildthecloud) [reply pkg/controllers/management/clusterprovisioner/provisioner.go]: AgentDeployed is fine

**Comment by cjellick** [comment]
@StrongMonkey can you look into rebasing this? if it isnt too complicated, can you do that?

**Comment by alena1108** [comment]
Closing in favor of https://github.com/rancher/rancher/pull/22461

**Pull request closed without merging**